test/data/javascript/tests.js

```diff
@@ -3347,6 +3347,51 @@ exports.test_data = {
             '}'
           ]
         },
+        {
+          comment: "#1095 - Return without semicolon followed by prefix on a new line",
+          input: [
+            'function x(){',
+            'return',
+            '++a',
+            '}',
+            '',
+            'while(true) {',
+            'return',
+            '--b',
+            '}'
+          ],
+          output: [
+            'function x() {',
+            '    return ++a',
+            '}',
+            '',
+            'while (true) {',
+            '    return --b',
+            '}'
+          ]
+        },
```

Comment:
Just to be clear this is a bug.  The output is functionally different from the input: 

```suggestion
          comment: "#1095 - Return without semicolon followed by prefix on a new line - BUG output should not pull items back to same line as return",
          input: [
            'function x(){',
            'return',
            '++a',
            '}',
            '',
            'while(true) {',
            'return',
            '--b',
            '}'
          ],
          output: [
            'function x() {',
            '    return ++a',
            '}',
            '',
            'while (true) {',
            '    return --b',
            '}'
          ]
        },
```

Reply:
It is functionally different in JS but if we think about a pure whitespace only formatter I would expect that it would get pulled back because it would not make sense to just have a return statement followed by anything. If this is not desired behaviour then I can fix the tests and the issue linked above.

Reply:
Regardless of what the formatter thinks or if the code doesn't make sense, if the formatter changes the functional behavior of the code it is not longer a formatter.  The input can be unreasonable and we still need to format it accurately.  Sorry.

Reply:
Understood! I will fix the issue.